src/pyqasm/modules/base.py

```diff
@@ -278,7 +279,10 @@ def depth(self):
         qasm_module = self.copy()
         qasm_module._qubit_depths = {}
         qasm_module._clbit_depths = {}
-        qasm_module.unroll()
+
+        # Unroll using any external gates that have been recorded for this
+        # module
+        qasm_module.unroll(external_gates = self._external_gates)
```

Comment:
Note: the design of this solution has been greatly influenced by this internal call: even though `unroll` may be called externally with the `external_gates` kwarg defined, internally here it has always been called without. Imho this can be problematic, hence the current approach that is being proposed.

Reply:
I guess `external_gates` is the only kwarg affecting the depth calculation so it is okay to use that here.